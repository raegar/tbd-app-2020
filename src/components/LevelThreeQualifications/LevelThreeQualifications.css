.form-background {
    width: 100%;
    background-color: #B2DDDF;
    display: inline-block;
    padding: 20px;
    border: 1px solid black;
    margin-bottom: 20px !important;
}

.centered-buttons {
    text-align: center;
}

.form-background h3 {
    margin-left: 60px;
}

.form-background h5 {
    margin-left: 60px;
}

.l3-form-container {
    text-align: center;
    margin: auto;
    padding: 10px;    width: 100%;
    left: 16%;
    padding-right: 0px !important;
    padding-left: 0px !important;
    margin-top: 2%;
    margin-bottom: 5%  !important;
}

.l3-mobile-form-container {
    text-align: center;
    margin: auto;
    width: 100%;
    padding: 10px;   
    left: 6.5%;
    margin-top: 10%;
    padding-right: 0px !important;
    padding-left: 0px !important;
}

.form-title {
    text-align: center;
}

.l3-form-right {
    position: relative;
    left: 30%;
    text-align: center !important;
    max-width: 40% !important;
}

.l3-mobile-form-right {
    position: relative;
    
    text-align: center !important;
    max-width: 100% !important;
}

.MobileTextArea
{
    max-width: 400px;
    max-height: 100px;
}

.mobiletextbox{
    margin: 0 auto;
}

@media only screen and (max-width: 600px) {
    .mobiletextbox {
        margin-left: 0%;
    }

    .form-background h3 {
        margin-left: 0px;
    }

    .form-background h5 {
        margin-top: 15px;
        margin-left: 0px;
    }
}
