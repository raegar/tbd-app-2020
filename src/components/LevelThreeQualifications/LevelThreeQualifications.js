import React, {useState} from "react";
import "./LevelThreeQualifications.css";
import L3Component from '../L3Component/L3Component';
import { isMobile } from 'react-device-detect';
import {Row, Col, Container} from "react-bootstrap";
import UCPButton from '../Buttons/UCPButton';
import ProgressBar from '../ProgressBar/ProgressBar';
import Header from '../Header/Header';
import Footer from '../Footer/Footer';

function LevelThreeQualifications() {
	const [qualificationsInfo, setQualificationsInfo] = useState([{}]);

	const className2 = isMobile ? "l3-mobile-form-container" : "l3-form-container";

	function saveSelectedData(counter) {
		let count = counter - 1;

		if (qualificationsInfo[count]) {
			global.ApplicationFormData.levelThreeQualifications.qual[count] = qualificationsInfo[count]["qualification"];
			global.ApplicationFormData.levelThreeQualifications.subject[count] = qualificationsInfo[count]["subject"];
			global.ApplicationFormData.levelThreeQualifications.grade[count] = qualificationsInfo[count]["grade"];
			global.ApplicationFormData.levelThreeQualifications.date[count] = qualificationsInfo[count]["year"];
		}

		console.log(global.ApplicationFormData);
	}

	function saveAllSelectedData() {
		for (let i = 1; i <= qualificationsInfo.length; i++) {
			if (qualificationsInfo[i - 1]) {
				saveSelectedData(i);
			}
		}
	}
	const completionRate = [
		{bgcolor: "#005C6E", completed:75 },
	];

  	return (
		  <div>
			<Header/>
			{completionRate.map((item, idx)=>(
				<ProgressBar key={idx} bgcolor={item.bgcolor} completed={item.completed}/>
			))}
  			<Container>
  				<div className={className2}>
  					<div className="form-background">
						<L3Component 
							setQualificationsInfo={setQualificationsInfo} 
							qualificationsInfo={qualificationsInfo} 
							saveSelectedData={saveSelectedData}
						/>
  					</div>
  					<Row id="buttonRow">
  						<Col className="centered-buttons">
  							<UCPButton 
							  	to='/LevelTwoQualifications'
  								primary="True"
  								className="mediumbutton"
  								buttonText="Go Back"
  							/>
  						</Col>
  						<Col className="centered-buttons">
  							<div id="confirmButton" onClick={saveAllSelectedData}>
  								<UCPButton
  									to="/DegreeQualifications"
  									primary="True"
  									className="mediumbutton"
  									buttonText="Confirm and Go"
  								/>
  							</div>
  						</Col>
  					</Row>
  				</div>
  			</Container>
			<Footer/>
		</div>
  	);
}

export default LevelThreeQualifications;
