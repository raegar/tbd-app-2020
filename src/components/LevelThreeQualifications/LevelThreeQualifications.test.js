import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import LevelThreeQualifications from './LevelThreeQualifications';
import { BrowserRouter as Router} from 'react-router-dom';

//cleans up 
afterEach(cleanup);

it("Renders LevelThreeQualifications", () => {
    const { asFragment } = render(<Router><LevelThreeQualifications/></Router>);
    expect(asFragment()).toMatchSnapshot();
});

/*it("Disclaimer Test Id", () => {
    const { getByTestId } = render(<Disclaimer/>);
    expect(getByTestId('ucpDisclaimer'));
});*/
