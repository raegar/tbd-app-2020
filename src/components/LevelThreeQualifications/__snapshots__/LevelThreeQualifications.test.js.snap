// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders LevelThreeQualifications 1`] = `
<DocumentFragment>
  <div>
    <div>
      <header
        class="Header"
      >
        <a
          href="/tbd-app-2020"
        >
          <img
            alt="UCP official logo"
            class="UCPLogoBlue"
            data-testid="ucpLogoHeader"
            src="Logo_White.png"
          />
        </a>
      </header>
      <div
        class="Line"
      />
    </div>
    <div
      style="height: 30px; width: 90%; background-color: rgb(178, 221, 223); border-radius: 50px; margin: 50px;"
    >
      <div
        style="height: 100%; width: 75%; background-color: rgb(0, 92, 110); border-radius: inherit; text-align: right;"
      >
        <span
          style="padding: 5px; color: rgb(255, 255, 255); font-weight: bold;"
        >
           75% 
        </span>
      </div>
    </div>
    <div>
      <div
        class="row"
      >
        <div
          class="col-md-4"
        >
          <div
            class="courseInfoBox undefined"
          >
            <div
              class="courseTitleRow row"
            >
              <h1
                class="courseInfoTitle"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-comment-dots fa-w-16 icon"
                  data-icon="comment-dots"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 32C114.6 32 0 125.1 0 240c0 49.6 21.4 95 57 130.7C44.5 421.1 2.7 466 2.2 466.5c-2.2 2.3-2.8 5.7-1.5 8.7S4.8 480 8 480c66.3 0 116-31.8 140.6-51.4 32.7 12.3 69 19.4 107.4 19.4 141.4 0 256-93.1 256-208S397.4 32 256 32zM128 272c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128 0c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128 0c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z"
                    fill="currentColor"
                  />
                </svg>
                  Course Info
              </h1>
            </div>
            <div
              class="row"
            >
              <button
                class="mediumbutton"
                data-testid="ucpButton"
                type="button"
              >
                Move UCAS Calculator to Here​
              </button>
            </div>
            <div
              class="uniPageLinkRow row"
            >
              <button
                class="smallbutton"
                data-testid="ucpButton"
                type="button"
              >
                Close Embed​
              </button>
              <div
                class="iframeContainer"
              >
                <iframe
                  class="iframeLink"
                  src="https://www.ucp.ac.uk/a-z-courses/"
                  title="Courses at UCP"
                />
              </div>
            </div>
            <div
              class="uniPageLinkRow row"
            >
              <a
                href="https://www.ucp.ac.uk/a-z-courses/"
              >
                Link to Course information page on UCP website
              </a>
            </div>
            <div
              class="ucasCalculatorRow row"
            >
              <p>
                <b>
                  UCAS Calculator:
                </b>
              </p>
              <div
                class="ucasCalculator"
              >
                <div>
                  <div
                    class="container"
                  >
                    <div
                      class="row"
                    >
                      <div
                        class="col"
                      >
                        <select
                          class="qualInputBox"
                          id="qualifications"
                        >
                          <option
                            disabled=""
                          >
                            Please select a qualification...
                          </option>
                          <option
                            class="gradeTypeHeader"
                            disabled=""
                          >
                            A Level and AS
                          </option>
                          <option
                            value="A Level and AS;A Levels"
                          >
                              A Levels
                          </option>
                          <option
                            value="A Level and AS;AS Levels"
                          >
                              AS Levels
                          </option>
                          <option
                            class="gradeTypeHeader"
                            disabled=""
                          >
                            BTEC National (RQF) (From 2016)
                          </option>
                          <option
                            value="BTEC National (RQF) (From 2016);Pearson BTEC Level 3 National Extended Certificate (first teaching from September 2016)"
                          >
                              Pearson BTEC Level 3 National Extended Certificate (first teaching from September 2016)
                          </option>
                        </select>
                      </div>
                    </div>
                    <div
                      class="row"
                    >
                      <div
                        class="col-md-2"
                      >
                        <div
                          class="mobiletextbox"
                        >
                          <label />
                          <input
                            class="subjectInputBox"
                            maxlength="30"
                            minlength="0"
                            placeholder="Subject (optional)"
                            type="text"
                          />
                        </div>
                      </div>
                      <div
                        class="col"
                      >
                        <br />
                      </div>
                      <div
                        class="col-md-2"
                      >
                        <br />
                        <select
                          class="Dropdown"
                          disabled=""
                        >
                          <option
                            value=""
                          />
                          <option
                            value="none"
                          >
                            ...
                          </option>
                        </select>
                      </div>
                    </div>
                    <div
                      class="row"
                    >
                      <div
                        class="col-md-6"
                      />
                      <div
                        class="col"
                      >
                        <span
                          class="qualPoints"
                          data-testid="qualPoints"
                        >
                          Points: 
                          <b>
                            0
                          </b>
                        </span>
                      </div>
                    </div>
                    <br />
                    <div
                      class="row"
                    >
                      <button
                        class="mediumbutton addQualButton"
                        data-testid="ucpButton"
                        type="button"
                      >
                        Add another qualification​
                      </button>
                    </div>
                    <span
                      class="totalQualPoints"
                    >
                      Total points: 
                      <b
                        data-testid="totalPoints"
                      >
                        0
                      </b>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="l3-form-container"
          >
            <div
              class="form-background"
            >
              <div
                class="l3-form-right"
              >
                <h3
                  class="form-title"
                >
                  Level 3 Qualifications
                </h3>
                <div
                  class="l3component"
                >
                  <h5>
                    Qualification : #1 
                  </h5>
                  <p>
                    (E.g. A Level, BTEC, Access)
                  </p>
                  <div
                    class="mobiletextbox"
                  >
                    <label
                      for="qualification1"
                    />
                    <input
                      id="qualification1"
                      maxlength="30"
                      minlength="0"
                      placeholder="Enter qualification #1"
                      type="text"
                      value=""
                    />
                  </div>
                  <h5>
                    Subject: (E.g. Business Studies)
                  </h5>
                  <div
                    class="mobiletextbox"
                  >
                    <label
                      for="subject1"
                    />
                    <input
                      id="subject1"
                      maxlength="30"
                      minlength="0"
                      placeholder="Enter subject #1"
                      type="text"
                      value=""
                    />
                  </div>
                  <h5>
                    Grade Achieved
                  </h5>
                  <div
                    class="mobiletextbox"
                  >
                    <label
                      for="grade1"
                    />
                    <input
                      id="grade1"
                      maxlength="30"
                      minlength="0"
                      placeholder="Enter grade #1"
                      type="text"
                      value=""
                    />
                  </div>
                  <h5>
                    Year Achieved
                  </h5>
                  <div
                    class="mobiletextbox"
                  >
                    <label
                      for="year1"
                    />
                    <input
                      id="year1"
                      max="9999-12-31"
                      maxlength="8"
                      min="1000-1-1"
                      minlength="0"
                      placeholder=""
                      type="date"
                      value=""
                    />
                  </div>
                  <br />
                  <h5>
                    Other L3 Qualifications
                  </h5>
                  <div
                    class="radio-buttons"
                  >
                    <h4 />
                    <p />
                    <div
                      class="radio-group"
                    >
                      <input
                        data-testid="qualificationYes"
                        id="qualificationYes"
                        name="l3qualifications"
                        type="radio"
                        value="yes"
                      />
                      <label
                        for="qualificationYes"
                      >
                        Yes
                      </label>
                    </div>
                  </div>
                  <br />
                  <div
                    class="row"
                  >
                    <div
                      class="col-5"
                    >
                      <button
                        class="smallbutton"
                        data-testid="ucpButton"
                        disabled=""
                        type="button"
                      >
                        Previous​
                      </button>
                    </div>
                    <div
                      class="col-2"
                    />
                    <div
                      class="col-5"
                    >
                      <button
                        class="smallbutton"
                        data-testid="ucpButton"
                        disabled=""
                        type="button"
                      >
                        Next​
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="row"
              id="buttonRow"
            >
              <div
                class="centered-buttons col"
              >
                <a
                  href="/LevelTwoQualifications"
                >
                  <button
                    class="mediumbutton"
                    data-testid="ucpButton"
                    primary="True"
                    type="button"
                  >
                    Go Back​
                  </button>
                </a>
              </div>
              <div
                class="centered-buttons col"
              >
                <div
                  id="confirmButton"
                >
                  <a
                    href="/DegreeQualifications"
                  >
                    <button
                      class="mediumbutton"
                      data-testid="ucpButton"
                      primary="True"
                      type="button"
                    >
                      Confirm and Go​
                    </button>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="ucpFooter"
    />
  </div>
</DocumentFragment>
`;
